packages/node_modules/@ciscospark/react-component-adaptive-card/src/index.js

```diff
@@ -17,11 +21,53 @@ class AdaptiveCard extends React.Component {
   componentDidMount() {
     if (Object.prototype.hasOwnProperty.call(this, 'nodeElement') && Object.prototype.hasOwnProperty.call(this.nodeElement, 'current')) {
       this.nodeElement.current.appendChild(
-        getAdaptiveCard(this.props.cards, this.props.displayName, this.props.sparkInstance, this.addChildNode)
+        getAdaptiveCard(this.state.cards, this.props.displayName, this.props.sparkInstance, this.addChildNode)
       );
     }
   }
 
+  static getDerivedStateFromProps(nextProps) {
```

Comment:
Can you explain the reason for using this lifecyle? [There's a lot of warnings from React team](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html) on using this lifecycle.

From my understanding of the code, it seem like this is only expected to run once (because of the `hasReplacedImagesInJSON` set once) but correct me if I'm misinterpreting

Reply:
As I mentioned earlier here:https://github.com/webex/react-widgets/pull/1034#discussion_r357017415 .   `getDerivedStateFromProps` will execute when props updated.

Reply:
Did you read the [alternatives to `getDeriveStateFromProps`](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops)? Would the suggestion of using `componentDidUpdate` be sufficient?

Reply:
used `componentDidUpdate` lifecycle

Reply:
Tried to use the `componentDidUpdate' life cycle method but props of last activity is not getting updated